EasyPost.Tests/ServicesTests/ShipmentServiceTest.cs

```diff
@@ -0,0 +1,193 @@
+                { "from_address", new Dictionary<string, object> { { "id", fromAddress.Id } } },
+                { "to_address", new Dictionary<string, object> { { "id", toAddress.Id } } },
+                { "parcel", new Dictionary<string, object> { { "id", parcel.Id } } }
+            });
+
+            Assert.IsType<Shipment>(shipment);
+            Assert.StartsWith("shp_", shipment.Id);
+            Assert.StartsWith("adr_", shipment.FromAddress.Id);
+            Assert.StartsWith("adr_", shipment.ToAddress.Id);
+            Assert.StartsWith("prcl_", shipment.Parcel.Id);
+            Assert.Equal("388 Townsend St", shipment.FromAddress.Street1);
+        }
+
+        [Fact]
+        [CrudOperations.Create]
+        [Testing.Parameters]
+        public async Task TestOneCallBuyShipmentWithCarbonOffset()
+        {
+            UseVCR("one_call_buy_shipment_with_carbon_offset");
+
+            Shipment shipment = await Client.Shipment.Create(Fixtures.OneCallBuyShipment, true);
+
+            Assert.NotNull(shipment.Fees);
+            bool carbonOffsetIncluded = shipment.Fees.Any(fee => fee.Type == "CarbonOffsetFee");
+            Assert.True(carbonOffsetIncluded);
+        }
+
+        [Fact]
+        [CrudOperations.Read]
+        [Testing.Function]
+        public async Task TestAll()
+        {
+            UseVCR("all");
+
+            ShipmentCollection shipmentCollection = await Client.Shipment.All(new Dictionary<string, object> { { "page_size", Fixtures.PageSize } });
+
+            List<Shipment> shipments = shipmentCollection.Shipments;
+
+            Assert.True(shipmentCollection.HasMore);
+            Assert.True(shipments.Count <= Fixtures.PageSize);
+            foreach (Shipment shipment in shipments)
+            {
+                Assert.IsType<Shipment>(shipment);
+            }
+        }
+
+        [Fact]
+        [CrudOperations.Read]
+        [Testing.Function]
+        public async Task TestGetLowestSmartrate()
+        {
+            UseVCR("get_lowest_smartrate");
+
+            Shipment shipment = await CreateBasicShipment();
+
+            // test lowest smartrate with valid filters
+            List<Smartrate> smartrates = await shipment.GetSmartrates();
+            Smartrate lowestSmartrate = ShipmentService.GetLowestSmartrate(smartrates, 2, SmartrateAccuracy.Percentile90);
+            Assert.Equal("Priority", lowestSmartrate.Service);
+            Assert.Equal(8.15, lowestSmartrate.Rate);
+            Assert.Equal("USPS", lowestSmartrate.Carrier);
+
+            // test lowest smartrate with invalid filters (should error due to strict delivery_days)
+            Assert.Throws<FilteringError>(() => ShipmentService.GetLowestSmartrate(smartrates, 0, SmartrateAccuracy.Percentile90));
+
+            // test lowest smartrate with invalid filters (should error due to bad delivery_accuracy)
+            // this test is not needed in the C# CL because it uses enums for the accuracy (can't pass in an incorrect value)
```

Comment:
These two lines should be removed

Reply:
I'm going to keep them to clarify to future travelers why we don't have a unit test for bad delivery accuracy